app/keymap-module/modules/modules.cmake

```
@@ -107,6 +114,21 @@ foreach(root ${BOARD_ROOT})
     endif()
 endforeach()
 
+if(EXISTS ${BOARD_DIR}/revision.cmake)
+    # Board provides revision handling.
+    include(${BOARD_DIR}/revision.cmake)
+elseif(BOARD_REVISION)
+    message(WARNING "Board revision ${BOARD_REVISION} specified for ${BOARD}, \
+                     but board has no revision so revision will be ignored.")
```

This wording is a bit confusing imo.  My suggestion might not be much better tho...
```suggestion
                     but board has no revision. Setting will be ignored.")
```

---

This piece is copied verbatim from the equivalent code in Zephyr, so I'd favor keeping it so searching later more easily correlates things.

---

Sounds like a great idea \:)